core/domain/classifier_services.py

```diff
@@ -124,6 +124,57 @@ def classify_string_classifier_rule(state, normalized_answer):
 
     return None
 
+def train(exploration):
```

Comment:
This needs more discussion. Is training going to be a synchronous or asynchronous process? That is, when the creator saves, do they have to wait for training to finish before the response comes back from the server and they are able to do anything else?

If it's an asynchronous process then I would expect the save operation to just enqueue the task for processing in a queue somewhere, rather than actually carry out the training.